plugin/vscode/extension.js

```diff
@@ -224,8 +230,8 @@ class DocumentLinter {
       this._pendingChanges.length = 0;
       this._state = DocumentLinterState.PARSER_LOADED;
       // END CRITICAL SECTION (no awaiting above)
-
-      let diags = this._parser.lint();
+      this._performanceWriter._fileName = this._document.fileName
+      let diags = this._parser.lint(this._performanceWriter);
```

Comment:
Minor: Why not pass the fileName as a parameter? This assignment looks weird, and it might accidentally get moved.

Reply:
[Addressed in latest commit]
this._fileName is now an argument without separating the path to have only the file name. The full path is logged now